### ISSUE TITLE
Twig reports incorrect file / line number if exception is thrown during the rendering of an include 

### ISSUE THREAD

**Issue by ken-spencer** [twigphp/Twig#1918]
If I throw an exception in a twig extension function run from an include, the reported file and line number are for the file that called the extension not the included file.  This can make debugging fairly tricky as we are looking for a needle in a haystack. 

Twig appears to be detecting the template by looping through  a debug_backtrace. This seems to result in the last template in the trace getting returned when I believe the first would be desired. 

If I make the following change in Error.php it appears to solve the problem: 

```
@@ -201,6 +201,7 @@
                 if (null === $this->filename || ($this->filename == $trace['object']->getTemplateName() && !$isEmbedContainer)) {
                     $template = $trace['object'];
                     $templateClass = get_class($trace['object']);
+                    break;
                 }
             }
         }
```

**Comment by stof** [comment]
Can you provide a reproducing testcase (and tell which version of Twig you are using) ?

**Comment by ken-spencer** [comment]
Sure. I am testing in * * 1.23.1, but this problem has existed for at least a year. 

## index.php

```
<?php


require __DIR__ . '/vendor/twig/twig/lib/Twig/Autoloader.php';
Twig_Autoloader::register();


$loader = new Twig_Loader_Filesystem(__DIR__);
$twig = new Twig_Environment($loader, [
    'cache' => __DIR__ . '/cache',
    'debug' => true,
    'auto_reload' => true,
    'strict_variables' => true,
]);

class Error extends Twig_Extension
{

    public function getFunctions()
    {
        return [
            'trigger_error' => new \Twig_Function_Method($this, 'triggerError'),
        ];                
    }

    function triggerError()
    {
        $error = new \Twig_Error_Runtime("This is an error");

        $str =  '<div>Error is in: ' . $error->getTemplateFile() . 'on line: ' . $error->getTemplateLine(). '</div>';
        $str .= '<div>Expecting:  Error is in: content.html.twigon line: 3 </div>';


        return $str;
    }

    public function getName()
    {
        return 'error';
    }

}

$twig->addExtension(new Error);
        

$template = $twig->loadTemplate('index.html.twig');
echo  $template->render([]);
```

## index.html.twig

```
<!DOCTYPE html>
<html>
    <head>
        <title>Test of error</title>
    </head>
    <body>
    <h1> get error </h1>
    {% include "content.html.twig" %}
    </body>
</html>
```

## content.html.twig

```
<div class="content">

{{ trigger_error() |raw }}

</div>
```

As you schoid be able to see the error is reported as being in index.html.twig rather than in content.html.twig. The diff I supplied above resolves the issue.

**Comment by stof** [comment]
wouldn't you diff break things for embeds though ?

**Comment by fabpot** [comment]
Will be fixed by #2935
